### DISCUSSION TITLE
Github Pages site disappeared, can't figure out why

### DISCUSSION THREAD

**Discussion by dobrian** [community/community#63868] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I have an online book hosted at <dobrian.github.io/cmp>. It was functioning fine for several years. Recently the site returns a 404 page saying "There isn't a GitHub Pages site here." I haven't touched the repository in at least a year or so. Try as I might, I can't figure out why the site in now unfindable. Insofar as possible, I have checked all settings and followed all instructions on how to make one's site visible (even though I had changed nothing). It's hosted in a /docs folder in my repository at <https://github.com/dobrian/cmp>. I've made the repository public, and will welcome any suggestions about how to restore the site to its prior visibility.

**Comment by melvincwng** [comment]
Hey @dobrian :wave:, I did a quick check - and your website is still visible at this domain: https://dobrian.github.io/cmp/docs/index.html

![image](https://github.com/community/community/assets/77479885/5c791b11-43f5-4893-8b32-7ffc1903b3d7)

I've also done a quick check on your code & repository. Since your files are in the /docs folder, that is why your website will appear in this URL - i.e https://dobrian.github.io/cmp/docs/index.html.

To break it down further, for your own understanding, why the URL of your website is as follows:
- **https://dobrian.github.io:** GitHub pages will create a domain name for you using your GitHub username (i.e. dobrian)
- **/cmp:** Name of your repository - 'cmp' in this case
- **/docs:** Name of folder where your HTML files are in
- **/index.html:** Name of your default HTML file

I hope this answers & clarifies your question. Take care, and have a good day ahead :smile:!

**Comment by dobrian** [comment]
Thanks much for your response. I understand all that you've written, and I have the same experience you do, of being able to see the site at the full address that you suggested. HOWEVER...

Previously the simple URL <https://dobrian.github.io/cmp> worked to access the site. I think the /docs folder was designated as a special default folder for Github to use for Pages. Or maybe at some point I needed to specify that folder, but I don't think so. Did that default use of the /docs folder change in Github's implementation of Pages? I see no way currently to specify that folder as the default for Pages. If you or anybody else knows how to make it the default folder, please let me know.

A new development is that the repository now (once again) lists https://dobrian.github.io/cmp as the Pages site (see attached screenshot), BUT...

Going to that URL just shows links to itself! It does not show the actual homepage, which as you noted, actually resides at <https://dobrian.github.io/cmp/docs/index.html>

Because there are many other sites that link to the original URL, I'd REALLY like to figure out how to restore the original, shorter URL to working properly.

**Reply by GhostOf0days** [reply]
Your website source code is in the docs folder not the root of the repo. That's why it's required in your URL. Also, in general Github Pages URLs are weird and can be unintuitive. I recommend you looking into changing the relative pathing in your index.html file (so say the index.html file was in the same path as another HTML file -> change the pathing to use "./htmlFileName.html" instead of "htmlFileName.html"), moving it to the root of your repo, and [using Netlify](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/) to deploy the website with a [_redirects](https://www.makeuseof.com/netlify-redirect-fail-error-fix/) folder. Github Pages is quite limited and not a really good way to deploy the website, especially if it's used for a class at UC Irvine. Github Pages is not scalable and only allows static websites/web apps. Other providers are Heroku, Railway, and Vercel. Railway and Vercel should be used when you need to scale. Use Railway when you need to centralize websites/web apps. Your use case should be small enough to warrant Netlify use. Netlify also allows you change the starting path of the website. You can check out the [Resilient Rhino repo](https://github.com/GhostOf0days/Resilient-Rhino/tree/main) on my profile for an example on set up if you go this route. You could also do the route of your index.html being at the root of your repo. Netlify, Heroku, Railway, and Vercel are all free and easy to use to deploy from a Github repo. They are also used widely in industry.

**Comment by dobrian** [comment]
Here are the screenshots I intended to attach to the previous post.
<img width="785" alt="Screenshot 2023-08-16 at 6 15 18 PM" src="https://github.com/community/community/assets/40784571/aea89230-2d1f-4975-a501-c4fd7092f9a2">
<img width="253" alt="Screenshot 2023-08-16 at 6 13 29 PM" src="https://github.com/community/community/assets/40784571/ed659cfa-ca72-47b0-8ddd-215bc6324501">

**Comment by dobrian** [comment, marked as answer]
Ah! I think I've found the solution. Under the repository (/cmp in my case), under Settings, under Pages, under Build and deployment, under Branch, there's a popup menu that allows to choose a default folder for the Pages site.
<img width="719" alt="Screenshot 2023-08-16 at 7 41 01 PM" src="https://github.com/community/community/assets/40784571/ad7fd5be-063f-4249-8348-43946e9ef616">
